dem_stitcher/stitcher.py

```diff
@@ -333,7 +337,7 @@ def reader(url):
     # Set driver in profile
     dem_profile['driver'] = driver
 
-    if (dem_name == 'glo_30') and fill_in_glo_30:
+    if (dem_name == 'glo_30') and fill_in_glo_30 and glo_90_missing_intersection:
         dem_arr, dem_profile = patch_glo_30_with_glo_90(dem_arr,
```

Comment:
There appears to be a bug in [line 302](https://github.com/ACCESS-Cloud-Based-InSAR/dem-stitcher/blob/194fa65d1c7369df9e4b9d575b4debbc963532fe/dem_stitcher/stitcher.py#L302), i.e. a call to `stitch_dem` within the `stitch_dem` function. I get the impression that was a vestige and you instead intended to include the call to `patch_glo_30_with_glo_90` there?

Reply:
This is correct.

For readability, I think it could be improved to update `fill_in_glo_30` to be `fill_in_glo_30 and glo_90_missing_intersection`.

To clarify, there are two cases:

1. Bounds are completely contained in `glo_90` tiles that are missing from `glo_30` - independent run of the `stitch_dem`
2. Bounds overlap the same "missing" tiles but also contain some tiles that are in `glo_30` - requires two runs of `stitch_dem` to be merged/patched.

It's organized the way it is so exceptions are not raised when calling `stitch_dem` twice; specifically, we do not get DEM coverage exceptions. 

I think reassigning the variable will make things clearer, too.